java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -132,6 +141,11 @@ public static Dataset commit(BufferAllocator allocator, String path,
   public static native Dataset commitAppend(String path, Optional<Long> readVersion,
       List<String> fragmentsMetadata);
 
+  /** Create a new Dataset Scanner. */
+  public Scanner newScan(ScanOptions options, Optional<String> filter) {
```

Comment:
nit: maybe add a few overloads for ease of use?
`public Scanner newScan(ScanOptions options)`
and
`public Scanner newScan()`